go/bazel.go

```diff
@@ -0,0 +1,158 @@
+	bazel string
+
+	// Path to the testing temp directory.
+	tmpDir string
+}
+
+// New instance of a TestingBazel will be created and the program will cd into
+// the appropriate working dir for that bazel client.
+func New() (TestingBazel, error) {
+	// Make a temporary dir in the OS specific tmpdir.
+	dir, err := ioutil.TempDir("", "testing")
+	if err != nil {
+		return nil, errors.New("Unable to make tempdir for bazel")
+	}
+
+	err = os.Chdir(dir)
+	if err != nil {
+		return nil, errors.New("Unable to make tempdir for bazel")
+	}
+
+	b := &testingBazel{
+		tmpDir: dir,
+	}
+	if !b.setAndUnpackBazel(fmt.Sprintf("build_bazel_bazel_%s/bazel", strings.Replace(bazelVersion, ".", "_", -1))) {
+		return nil, errors.New("Unable to find Bazel")
+	}
+
+	return b, nil
+}
+
+func (b *testingBazel) setAndUnpackBazel(bazel string) bool {
+	bazel = b.Rlocation(bazel)
+	if _, err := os.Stat(bazel); !os.IsNotExist(err) {
+		b.bazel = bazel
+		// Unpack Bazel by running a command on it.
+		b.RunBazel([]string{"help"})
+		return true
+	}
+	return false
+}
+
+// ScratchFile creates a new file at the path you provided.
+func (b *testingBazel) ScratchDir(p string) error {
+	return os.MkdirAll(p, 0755)
+}
+
+// ScratchFile creates a new file at the path you provided. If the file is in a
+// directory that may not exist, use ScratchDir to make the directory. To set a
+// mode on the file use ScratchFileWithMode.
+func (b *testingBazel) ScratchFile(p, content string) error {
+	return b.ScratchFileWithMode(p, content, 0644)
+}
+
+// ScratchFile creates a new file at the path you provided with a specified
+// mode (useful for setting executable bit). If the file is in a directory that
+// may not exist, use ScratchDir to make the directory.
+func (b *testingBazel) ScratchFileWithMode(p, content string, mode os.FileMode) error {
+	if err := ioutil.WriteFile(p, []byte(content), mode); err != nil {
+		return err
+	}
+	return nil
+}
+
+// GetBazel returns the path to the Bazel binary that is available.
+func (b *testingBazel) GetBazel() string {
+	return b.bazel
+}
+
+func (b *testingBazel) RunBazel(args []string) (int, string, string) {
+	find := exec.Command("ls", "-laR")
+	find.Stdout = os.Stdout
+	find.Run()
+
+	cmd := exec.Command(b.bazel, args...)
+
+	var stdout, stderr bytes.Buffer
+	cmd.Stdout = &stdout
+	cmd.Stderr = &stderr
+
+	if err := cmd.Run(); err != nil {
+		if exitErr, ok := err.(*exec.ExitError); ok {
+			status := exitErr.Sys().(syscall.WaitStatus)
+			return status.ExitStatus(), stdout.String(), stderr.String()
+		}
+	}
+
+	// When there isn't an error we know it exited successfully
+	return 0, stdout.String(), stderr.String()
+}
+
+// IsWindows returns true/false depending on if you're on Windows.
+func (b *testingBazel) IsWindows() bool {
+	// TODO: Move this into a linux/darwin specific file.
+	return false
+}
+
+// Rlocation Returns the absolute path to a runfile.
+func (b *testingBazel) Rlocation(runfile string) string {
```

Comment:
`github.com/bazelbuild/rules_go/go/tools/bazel.Runfile` should do the same thing.

Reply:
Done